### PR TITLE
Generate settings manifest

### PR DESCRIPTION
**Pull request by bion**
### Description

Adds python code for generating the `ServerSettings.java` file described in [admin settings panel design](https://docs.google.com/document/d/1FK5YLJd6jVu1O1uuu2OOR2mXTs3yVEXkzyfjVwnkz8s/edit).

Does not consume the generated code. The next PR will though.

The logic for retrieving config values won't work until https://github.com/civiform/civiform/pull/4982 is merged.

#### General

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release >
- [x] Created tests which fail without the change (if possible)

### Issue(s) this completes

Contributes to https://github.com/civiform/civiform/issues/4948

### CODE REVIEW

File: server/app/services/settings/SettingsManifest.java [hunk 1]
```diff
@@ -0,0 +1,1344 @@
...
+  }
+
+  /** Small logo for the civic entity used on the login page. */
+  public String getWhitelabelSmallLogoUrl() {
+    return getString("WHITELABEL_SMALL_LOGO_URL");
+  }
+
+  /** Logo with civic entity name used on the applicant-facing program index page. */
+  public String getWhitelabelLogoWithNameUrl() {
+    return getString("WHITELABEL_LOGO_WITH_NAME_URL");
```
Comment (rockycodes) [inline server/app/services/settings/SettingsManifest.java:48]: nitpick (non-blocking): regenerate before merging to remove this var that was removed in your other pr.

**Review by rockycodes — APPROVED** [review]
This looks good to me! It's such a cool setup. I am not the most well versed in Java and Python, so recommend getting another set of eyes if you would like more feedback/suggestions on your code. But overall the logic/setup is clear and makes sense to me, just a couple of nitpicks.

File: server/app/services/settings/AbstractSettingsManifest.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+  protected ImmutableList<String> getListOfStrings(SettingDescription settingDescription) {
+    return ImmutableList.copyOf(config.getStringList(getHoconName(settingDescription)));
+  }
+
+  protected ImmutableList<String> getListOfStrings(String variableName) {
+    return ImmutableList.copyOf(config.getStringList(getHoconName(variableName)));
+  }
+
+  private static String getHoconName(String variableName) {
```
Comment (reemaka) [inline server/app/services/settings/AbstractSettingsManifest.java:53]: What's "Hocon"?

File: env-var-docs/parser-package/src/env_var_docs/settings_manifest.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+import dataclasses
+from jinja2 import Environment, PackageLoader, select_autoescape
+from env_var_docs.parser import Variable, Node, Group, NodeParseError, visit
+import env_var_docs.errors_formatter
+import typing
+import string
+import os
+import sys
+from io import StringIO
```
[the remark this answers is not in the document]
Comment (bion) [reply env-var-docs/parser-package/src/env_var_docs/settings_manifest.py:13]: Done

File: env-var-docs/parser-package/src/env_var_docs/settings_manifest.py [hunk 4]
```diff
@@ -0,0 +1,210 @@
+"""Generates a settings manifest in Java for use in the CiviForm server based
```
[the remark this answers is not in the document]
Comment (bion) [reply env-var-docs/parser-package/src/env_var_docs/settings_manifest.py:1]: Done

File: env-var-docs/parser-package/src/env_var_docs/settings_manifest.py [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+
+    return Config(docs, local)
+
+
+def main():
+    config = make_config()
+    with open(config.docs_path) as docs_file:
+        manifest, parse_errors = generate_manifest(docs_file)
+
+        if len(parse_errors) != 0:
```
[the remark this answers is not in the document]
Comment (bion) [reply env-var-docs/parser-package/src/env_var_docs/settings_manifest.py:55]: Done

File: env-var-docs/parser-package/src/env_var_docs/settings_manifest.py [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+            msg = f"{config.docs_path} is invalid:\n"
+            msg += env_var_docs.errors_formatter.format(parse_errors)
```
[the remark this answers is not in the document]
Comment (bion) [reply env-var-docs/parser-package/src/env_var_docs/settings_manifest.py:57]: Done

File: env-var-docs/parser-package/src/env_var_docs/settings_manifest.py [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+    out = StringIO()
+
+    out.write("ImmutableMap.of(")
+
+    groups = ", ".join(
+        [
+            f'"{group.group_name}", {render_group(group)}'
+            for group in root_group.sub_groups
+        ])
+    if groups != "":
```
[the remark this answers is not in the document]
Comment (bion) [reply env-var-docs/parser-package/src/env_var_docs/settings_manifest.py:79]: Done

File: server/app/services/settings/SettingsManifest.java (hunk 1 above)
Comment (bion) [reply server/app/services/settings/SettingsManifest.java:48]: Done

File: env-var-docs/parser-package/tests/settings_manifest_test.py [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+                        regex=None,
+                        regex_tests=None,
+                        mode=Mode.HIDDEN)
+            })
+
+        result = render_sections(root_group)
+
+        self.assertEqual(
+            result,
+            """ImmutableMap.of("Test section", SettingsSection.create("Test section", "Fake section for testing.", ImmutableList.of(SettingsSection.create("Test subsection", "Fake subsection for testing", ImmutableList.of(), ImmutableList.of(SettingDescription.create("SUBSECTION_VARIABLE", "Fake subsection variable for testing", SettingType.STRING)))), ImmutableList.of()), "ROOT", SettingsSection.create("ROOT", "Top level vars", ImmutableList.of(), ImmutableList.of(SettingDescription.create("STRING_VARIABLE", "Fake string variable for testing", SettingType.STRING))))"""
```
[the remark this answers is not in the document]
Comment (bion) [reply env-var-docs/parser-package/tests/settings_manifest_test.py]: Done

File: env-var-docs/parser-package/src/env_var_docs/settings_manifest.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (reemaka) [reply env-var-docs/parser-package/src/env_var_docs/settings_manifest.py:15]: Lol the link was supposed to be: https://google.github.io/styleguide/pyguide.html#313-imports-formatting, oops

**Review by reemaka — APPROVED** [review]

File: env-var-docs/parser-package/src/env_var_docs/settings_manifest.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (bion) [reply env-var-docs/parser-package/src/env_var_docs/settings_manifest.py:15]: I was wondering 😆

**Pull request merged**